central/policycategory/datastore/datastore_impl.go

```diff
@@ -32,11 +35,81 @@ var (
 )
 
 type datastoreImpl struct {
-	storage  store.Store
-	indexer  index.Indexer
-	searcher search.Searcher
+	storage              store.Store
+	indexer              index.Indexer
+	searcher             search.Searcher
+	policyCategoryEdgeDS datastore.DataStore
+	categoryMutex        sync.Mutex
+}
+
+func (ds *datastoreImpl) SetPolicyCategoriesForPolicy(ctx context.Context, policyID string, categoryNames []string) error {
+	ds.categoryMutex.Lock()
+	defer ds.categoryMutex.Unlock()
+
+	results, err := ds.policyCategoryEdgeDS.SearchEdges(ctx, searchPkg.NewQueryBuilder().AddExactMatches(searchPkg.PolicyID, policyID).ProtoQuery())
+	if err != nil {
+		return err
+	}
+	existingCategoryNames := make([]string, 0, len(results))
+	for _, result := range results {
+		existingCategoryNames = append(existingCategoryNames, result.Name)
```

Comment:
This seems incorrect because edge `result.Name` is ID

```
func convertOne(obj *storage.PolicyCategoryEdge, result *search.Result) *v1.SearchResult {
	return &v1.SearchResult{
		Category:       v1.SearchCategory_POLICY_CATEGORY_EDGE,
		Id:             obj.GetId(),
		Name:           obj.GetId(),
		FieldToMatches: search.GetProtoMatchesMap(result.Matches),
		Score:          result.Score,
	}
}
```

Reply:
or `existingCategoryNames` seems to be a misleading name